app/controllers/candidate_interface/apply_controller.rb

```diff
@@ -0,0 +1,18 @@
+module CandidateInterface
+  class ApplyController < CandidateInterfaceController
+    def show
+      @provider_code = params.fetch(:providerCode)
+      @course_code = params.fetch(:courseCode)
+
+      @course = Course
+        .where(recruitment_cycle_year: '2020')
+        .where(provider_code: @provider_code)
+        .find(@course_code)
+        .first
+    rescue JsonApiClient::Errors::NotFound
+      render :not_found, status: :not_found
+    rescue StandardError
```

Comment:
What are we expecting here? It would be good to be specific about the errors that we rescue.

Reply:
Basically any 50x-ish error which are caught by [JsonApiClient::Errors::UnexpectedStatus](https://www.rubydoc.info/gems/json_api_client/0.2.2/JsonApiClient/Errors/UnexpectedStatus) I believe, so can change to that.